[charts] Improve the management of the text (legend only) (#10138)
mui/mui-x: docs/data/charts/legend/legend.md

{{"demo": "BasicLegend.js"}}

## Customization

### Position

The legend can either be displayed in a `'column'` or `'row'` layout controlled with the `direction` property.

It can also be moved with the `position: { vertical, horizontal }` property which defines how the legend aligns within the SVG.

- `vertical` can be `'top'`, `'middle'`, or `'bottom'`.
- `horizontal` can be `'left'`, `'middle'`, or `'right'`.

> You can add some space to a given `position` with a `padding` property which can be either a number or an object `{ top, bottom, left, right }`.
> This `padding` will add space between the SVG borders and the legend.

Reviewer: I am not entirely sure about the phrasing of 'a given position'. It makes sense to me, but I'm not sure it is intelligible at a first glance. I don't have a better suggestion for this though 🙈 🤔 
```suggestion
You can add spacing to a given `position` with the `padding` property, which can be either a number or an object `{ top, bottom, left, right }`.
This `padding` will add space between the SVG borders and the legend.
```

> Defaults are such that the legend is placed on top of the charts.

Reviewer: ```suggestion
By default, the legend is placed above the charts.
```

{{"demo": "LegendPositionNoSnap.js", "hideToolbar": true, "bg": "inline"}}

### Dimensions

> Inside the legend, the mark can be customized with `itemMarkWidth` and `itemMarkHeight` which are the width/height in pixels.

Reviewer: ```suggestion
Inside the legend, you can customize the pixel value of the width and height of the mark with the `itemMarkWidth` and `itemMarkHeight` props.
```

You also have access to the `markGap` which is the gap between the mark and its label given in pixels.

> And `itemGap` which is the gap between two legend items.

Author: Adding the demo allowed me to spot on mark/label placement. Rows were overlapping when `itemGap` was small.

Now it's fixed `itemGap=0` means every think touch its neighbors
[legendbug.mp4.webm](https://github.com/mui/mui-x/assets/45398769/2f399d38-d987-442c-b898-abfee0c5f135)

> You also have access to the `markGap` which is the gap between the mark and its label given in pixels.
> And `itemGap` which is the gap between two legend items.

Reviewer: ```suggestion
You can also access the `markGap` prop to change the gap between the mark and its label, or the  `itemGap` to change the gap between two legend items. Both props impact the values defined in pixels.
```

> ### Label styling

Reviewer 2: Would it make sense to have this on the [styling page](https://deploy-preview-10138--material-ui-x.netlify.app/x/react-charts/styling/) since it doesn't apply only to legends?

Author: What about keeping a "Label styling" section on all pages that use text. 

For example in axis page it will have similar content explaining how to customize axis labels and ticks label. Saying what's the name of the styling property and how to access it.

In the styling page, we could have an explanation about which properties are supported. For example, the text baseline, the alignement, or the rotation might be slightly different than standard CSS.

To break line in legend labels, use the special character `\n`.

To customize its style, you should not use CSS. Instead pass a styling object to the `labelStyle` property.

> The `labelStyle` property is needed to measure text size, and then place legend items at the correct position.

Author: I was hesitating to put it into an `info` section. It's not mandatory information, but it's good to know why this object is here and what you might break if you don't use it
